scripts/agent-provision.sh

```
@@ -0,0 +1,132 @@
+  echo "→ Granting passwordless sudo to ${AGENT_USER} (--with-sudo)"
+  printf '%s ALL=(ALL) NOPASSWD:ALL\n' "$AGENT_USER" > "/etc/sudoers.d/90-agent-${AGENT_USER}"
+  chmod 0440 "/etc/sudoers.d/90-agent-${AGENT_USER}"
+else
+  echo "→ Leaving ${AGENT_USER} WITHOUT sudo (rootless agent space)"
+fi
+
+# ── 3. Shared Homebrew (glibc only; Alpine uses apk in the apply phase) ───────
+if [ "$OS_FAMILY" != "alpine" ]; then
+  if [ ! -x /home/linuxbrew/.linuxbrew/bin/brew ]; then
+    echo "→ Installing Homebrew (shared prefix) as ${AGENT_USER}"
+    su - "$AGENT_USER" -c 'NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"'
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)